spec/features/observer_spec.rb

```diff
@@ -39,7 +39,7 @@
   it "allows a user to add a reason when adding an observer" do
     reason = "is the archbishop of banterbury"
     proposal = create(:proposal)
-    observer = create(:user)
+    observer = create(:user, client_slug: nil)
```

Comment:
I could have clarified with a comment in the test... ;)

The reason is that in order for `Proposal.client` to match, both values must be null. Since the Proposal has no attached ClientData, its value is null. So User must use explicitly identical value.